plugins/unmanaged_files/unmanaged_files_inspector.rb

```diff
@@ -213,6 +212,57 @@ def inspect(filter, options = {})
 
     scope.scope_file_store = file_store_tmp
 
+    file_filter = filter.element_filter_for("/unmanaged_files/files/name").dup if filter
+    file_filter ||= ElementFilter.new("/unmanaged_files/files/name")
+    file_filter.add_matchers("=", @description.store.base_path)
+
+    # Add a recursive pendant to each ignored element
+    file_filter.matchers.each do |operator, matchers|
+      file_filter.add_matchers(operator, matchers.map { |entry| File.join(entry, "/*") })
+    end
+
+    helper = MachineryHelper.new(@system)
+    if helper_usable?(helper, options)
+      begin
+        helper.inject_helper
+        helper.run_helper(scope)
+      ensure
+        helper.remove_helper
+      end
+      scope.extracted = false
+
+      scope.files.delete_if { |f| file_filter.matches?(f.name) }
+
+      @description["unmanaged_files"] = scope
+    else
+      run_inspection(file_filter, options, do_extract, file_store_tmp, file_store_final, scope)
+    end
+  end
+
+  def helper_usable?(helper, options)
```

Comment:
Unfortunately not. That's why the code wasn't in a separate method but part of the inspect method in my branch. If you also cannot come-up with a better name, I'm fine with it as it is right now.